core/src/main/java/io/micronaut/core/convert/ConversionService.java

```diff
@@ -15,49 +15,24 @@
  */
 package io.micronaut.core.convert;
 
+import io.micronaut.core.annotation.Nullable;
 import io.micronaut.core.convert.exceptions.ConversionErrorException;
 import io.micronaut.core.type.Argument;
-import io.micronaut.core.annotation.Nullable;
+
 import java.util.Optional;
-import java.util.function.Function;
 
 /**
  * A service for allowing conversion from one type to another.
  *
- * @param <Impl> The type
  * @author Graeme Rocher
  * @since 1.0
  */
-public interface ConversionService<Impl extends ConversionService> {
+public interface ConversionService {
```

Comment:
I'm concerned about the removal of the generic type argument here as it likely to break a lot of end user code. Is this a necessary change? What is the justification?

Reply:
It was only used for the mutable `add` methods, now those methods are gone so the generic parameter is unused.

Reply:
It does mean though that any place a user injected `ConversionService<?>` will break which might be a significant breaking change. Thoughts @sdelamo ?

Reply:
Users might need to modify their code anyway if relying on the shared conversion service.

Reply:
Our docs use `ConversionService<?>` in examples. Still I never liked that generic type argument, so my +1 to remove it if it's not needed